src/overlays/actors/ovl_En_Famos/z_en_famos.c

```diff
@@ -28,133 +62,811 @@ const ActorInit En_Famos_InitVars = {
+                this->cratorDespawnTimer = 70;
+                EnFamos_SetupDeathSlam(this);
+            } else {
+                this->cratorDespawnTimer = 20;
+                EnFamos_SetupFinishAttack(this);
+            }
+        } else {
+            this->emblemCollider.base.acFlags |= AC_ON;
+            EnFamos_SetupAttackRebound(this);
+        }
+    } else {
+        func_800B9010(&this->actor, NA_SE_EN_LAST1_FALL_OLD - SFX_FLAG);
+    }
+}
+
+/**
+ * Summary: Attack has hit; Wait for animation to finish before recovering from attack.
+ */
+void EnFamos_SetupFinishAttack(EnFamos* this) {
+    Animation_PlayOnce(&this->skelAnime, &gFamosIdleAnim);
+    SkelAnime_Update(&this->skelAnime);
+    this->emblemCollider.base.acFlags |= AC_ON;
+    this->stateTimer = 3;
+    this->actor.speedXZ = 0.0f;
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_EXPLOSION);
+    this->actionFunc = EnFamos_FinishAttack;
+}
+
+void EnFamos_FinishAttack(EnFamos* this, GlobalContext* globalCtx) {
+    if (this->stateTimer == 0) {
+        this->collider2.base.atFlags &= ~AT_ON;
+    }
+
+    this->stateTimer--;
+    if (SkelAnime_Update(&this->skelAnime)) {
+        EnFamos_SetupAttackRebound(this);
+    }
+}
+
+void EnFamos_SetupAttackRebound(EnFamos* this) {
+    this->actor.world.rot.x = 0x4000;
+    this->actionFunc = EnFamos_AttackRebound;
+    this->actor.speedXZ = 0.0f;
+}
+
+void EnFamos_AttackRebound(EnFamos* this, GlobalContext* globalCtx) {
+    Math_StepToF(&this->actor.speedXZ, 5.0f, 0.3f);
+    if (this->actor.speedXZ > 1.0f) {
+        if (ABS_ALT(this->flipRot) > 0x4000) {
+            func_800B9010(&this->actor, NA_SE_EN_FAMOS_FLOAT_REVERSE - SFX_FLAG);
+        } else {
+            func_800B9010(&this->actor, NA_SE_EN_FAMOS_FLOAT - SFX_FLAG);
+        }
+    }
+
+    if (this->baseHeight < this->actor.world.pos.y || this->actor.bgCheckFlags & 0x10) { // touching ceiling
+        this->actor.speedXZ = 0.0f;
+        EnFamos_SetupChase(this);
+    }
+}
+
+/**
+ * Summary: Looking around for player.
+ */
+void EnFamos_SetupScanForPlayer(EnFamos* this) {
+    this->stateTimer = 60;
+    this->actionFunc = EnFamos_ScanForPlayer;
+    this->actor.speedXZ = 0.0f;
+}
+
+void EnFamos_ScanForPlayer(EnFamos* this, GlobalContext* globalCtx) {
+    EnFamos_UpdateBobbingHeight(this);
+    this->stateTimer--;
+
+    if (EnFamos_IsPlayerSeen(this, globalCtx)) {
+        EnFamos_SetupAlert(this);
+    } else if (this->stateTimer == 0) {
+        EnFamos_SetupTurnHome(this);
+    } else {
+        this->actor.shape.rot.y = (s32)(Math_SinS(this->stateTimer * 0x888) * 8192.0f) + this->actor.world.rot.y;
+    }
+}
+
+/**
+ * Summary: Famos was upside down and hit the ground headfirst. This kills the Famos.
+ */
+void EnFamos_SetupDeathSlam(EnFamos* this) {
+    this->emblemCollider.base.acFlags &= ~AC_ON;
+    this->stateTimer = 20;
+    this->actor.speedXZ = 0.0f;
+    Actor_SetColorFilter(&this->actor, 0x4000, 0xFF, false, 20);
```

Comment:
```suggestion
    Actor_SetColorFilter(&this->actor, 0x4000, 255, 0, 20);
```
The fourth argument is not boolean, for example
```
src/overlays/actors/ovl_En_Firefly/z_en_firefly.c:        Actor_SetColorFilter(&this->actor, 0x4000, 255, 0x2000, 40);
src/overlays/actors/ovl_En_Firefly/z_en_firefly.c:        Actor_SetColorFilter(&this->actor, 0x4000, 255, 0, 40);
src/overlays/actors/ovl_En_Firefly/z_en_firefly.c:        Actor_SetColorFilter(&this->actor, 0, 255, 0x2000, this->timer);
src/overlays/actors/ovl_En_Firefly/z_en_firefly.c:        Actor_SetColorFilter(&this->actor, 0, 255, 0, this->timer);
```
(applies elsewhere)

Reply:
Weird, not sure how I got the impression that was a bool